packages/react-dev-utils/webpackHotDevClient.js

```diff
@@ -28,9 +28,18 @@ var ErrorOverlay = require('react-error-overlay');
 ErrorOverlay.startReportingRuntimeErrors({
   launchEditorEndpoint: launchEditorEndpoint,
   onError: function() {
-    // TODO: why do we need this?
-    if (module.hot && typeof module.hot.decline === 'function') {
-      module.hot.decline();
+    // Ensure HotModuleReplacementPlugin is active
+    if (module.hot && typeof module.hot.addStatusHandler === 'function') {
+      // When a run time error occurs, it does not make sense to continue HMR;
+      // as application state may be corrupted.
+      // So, next time we check for updates, simply reload the page (to
+      // abort the process).
+      // See https://github.com/facebookincubator/create-react-app/issues/3096
+      module.hot.addStatusHandler(function(status) {
+        if (status === 'check') {
+          window.location.reload();
```

Comment:
Instead of doing this here, we may just want to replace all of this with a boolean we set to true, then check here:
https://github.com/facebookincubator/create-react-app/blob/fcb6dc55578e089e9f62f6e6c12cb9b60a57fee5/packages/react-dev-utils/webpackHotDevClient.js#L229-L233

/cc @gaearon

Reply:
Effectively, `if (err || !updatedModules || hadRuntimeError) {`.